src/development/platform-integration/ios/app-extensions.md

```diff
@@ -0,0 +1,211 @@
+   To learn more about the generated code and the SDKs for each
+   extension point, check out the resources in
+   [Apple's documentation][].
+
+## How do Flutter apps interact with App Extensions? 
+Flutter apps interact with app extensions using the same
+techniques as UIKit or SwiftUI apps.
+The containing app and the app extension don't communicate directly.
+These components read and write to shared resources or
+use higher-level APIs to communicate with each other.
+
+### Using higher-level APIs
+Some extension points have APIs to configure or update the app
+extension.
+
+For example, the [Core Spotlight][] framework indexes your app 
+allowing users to search from Spotlight and Safari. The
+[WidgetKit][] framework can trigger an update of your home screen
+widget.
+
+For APIs that don't display a UI, use a plugin to make calls to and
+from your Flutter app. To find plugins that wrap app extension APIs,
+check out [Leveraging Apple's System APIs and Frameworks][] or
+search [pub.dev][].
+
+### Sharing resources
+To share resources between your Flutter app and your app extension, put
+the `Runner` app target and the extension target in the same
+[App Group][].
+
+To add a target to an App Group:
+
+1. Open the target settings in Xcode.
+1. Navigate to the **Signing & Capabilities** tab.
+1. Select **+ Capability** then **App Groups**.
+1. Select or create the App Group.
+
+{{site.alert.note}}
+  You must be signed in to your Apple Developer account.
+{{site.alert.end}}
+
+{% include docs/app-figure.md
+image="development/platform-integration/app-extensions/xcode-app-groups.png" %}
+
+When two targets belong to the same App Group, they can read and write
+data to the same container using one of the following
+options:
+
+- **Key/value:** Use the [`shared_preference_app_group`][]
+  plugin to read or write to `UserDefaults` within the same App Group.
+- **File:** Use the App Group container path from the
+  [`path_provider`][] plugin to [read and write files][].
+- **Database:** Use the App Group container path from
+  the [`path_provider`][] plugin to create a database with the
+  [`sqflite`][] plugin.
+
+### Background updates
+Background tasks provide a means to update your extension through code
+regardless of the status of your app.
+
+To schedule background work from your Flutter app, use the
+[`workmanager`][] plugin.
+
+### Deep linking
+You might want to direct users from an app extension to a
+specific page in your Flutter app.
+To have a URL open a specified route in your app, you can use
+[Deep Linking][].
+
+## Creating app extension UIs with Flutter
+Some app extensions display a user interface.
+For example, iMessage extensions allow users to access your app's
+content directly from the **Messages** app.
+
+<figure class="site-figure {{include.class}}">
+    <div class="site-figure-container">
+        <img src='/assets/images/docs/development/platform-integration/app-extensions/imessage-extension.png'
```

Comment:
I couldn't figure it out which is why I used an img HTML tag instead of liquid / markdown. I changed the height to be a bit smaller, but let me know if there's a better approach!

Reply:
I meant the size of the image itself is 1.29 MB which seems very large to check into this repository and to load on the page, not that the size is too large on the screen (although I didn't deploy this so I'm not sure about that).  
https://github.com/flutter/website/issues/6487
I swear we used to have `convert` imagemagick instructions with compression suggestions, but I can't find it anywhere so maybe I dreamed it.

Reply:
I can reduce the size locally, thanks for the catch!